axis.go

```diff
@@ -424,26 +429,88 @@ var _ Ticker = LogTicks{}
 // Ticks returns Ticks in a specified range
 func (LogTicks) Ticks(min, max float64) []Tick {
 	var ticks []Tick
+	var labels []float64
+	var new_labels []float64
+	labels = nil
 	val := math.Pow10(int(math.Floor(math.Log10(min))))
+
 	if min <= 0 {
 		panic("Values must be greater than 0 for a log scale.")
 	}
-	prec := precisionOf(max)
+
 	for val < max*10 {
 		for i := 1; i < 10; i++ {
 			tick := Tick{Value: val * float64(i)}
 			if i == 1 {
-				tick.Label = formatFloatTick(val*float64(i), prec)
+				labels = append(labels, val*float64(i))
 			}
+			//Makes a list of small ticks
 			ticks = append(ticks, tick)
 		}
 		val *= 10
 	}
-	tick := Tick{Value: val, Label: formatFloatTick(val, prec)}
-	ticks = append(ticks, tick)
+
+	new_labels = adjustPrecision(labels)
+
+	for j := 0; j < len(labels); j++ {
+		//Adds big ticks to the list of small ones
+		ticks = append(ticks, Tick{Value: labels[j], Label: formatFloatTick(new_labels[j], -1)})
+	}
 	return ticks
 }
 
+func adjustPrecision(elements []float64) []float64 {
+	const maxExp = 308 //precision of the MaxFloat64 which we take as the maximum one here
+	for i := 1; i < maxExp; i++ {
+		var result []float64
+		i10 := math.Pow10(i)
+
+		for _, v := range elements {
+			var new_element float64
+			vpow := v * i10
+			//As within 'if' statement vpow is turned to integer, it cannot exceed the MaxInt64. If it does, another rounding technique is used.
+			if vpow < math.MaxInt64 {
+				if v > 0 {
+					new_element = float64(int(vpow)) / i10
+				} else {
+					new_element = 0 - float64(int(math.Abs(v)*i10))/i10
+				}
+				result = append(result, new_element)
+			} else {
+				new_element, err := strconv.ParseFloat(formatFloatTick(v, i), 64)
+				if err != nil {
+					panic(err)
+				}
+				result = append(result, new_element)
+			}
+		}
+		//checks whether the result array has duplicates. If not, the right level of precision is found, if it does, one more digit to the precision level is added and the check runs again.
```

Comment:
This comment is not necessary since the code speaks for itself. If you are troubled, a nicer way to make the explanation clearer would be to write

```
    if !hasDuplicates(result) {
        return result
    }
```

then (since `removeDuplicates`'s return value is not used for anything else beyond testing for whether an element has been removed) `removeDuplicates` can be rewritten as `hasDuplicates`:

```
// hasDuplicates returns whether the sorted slice f has a duplicate element.
// Standard IEEE 754 NaN equality is used since tick values are not expected
// to contain NaN values.
func hasDuplicates(f []float64) bool {
    for i, v := range f[1:] {
        if v == f[i] {
            return true
        }
    }
    return false
}
```

This improves the code, the performance, and the clarity.